### PR TITLE
Fixes #16083: Big memory usage when fetching/writing ndoe configuration and expected reports

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/16083

### CODE REVIEW

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
Commit modified

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ExpectedReportsJdbcRepository.scala [hunk 1]
```diff
@@ -211,47 +231,64 @@ class UpdateExpectedReportsJdbcRepository(
   }
 
 
-  def saveNodeExpectedReports(configs: List[NodeExpectedReports]): Box[List[NodeExpectedReports]] = {
-
+  def saveNodeExpectedReports(configs: List[NodeExpectedReports]): Box[Seq[NodeExpectedReports]] = {
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ExpectedReportsJdbcRepository.scala]: this fails with 
[2019-10-31 15:03:21] ERROR policy.generation - Error when updating policy, reason was: Error when saving expected reports <- Batch entry 0 
                         insert into nodes_info (config_ids, node_id) values ('[{"2019-10-31T14:52:53.287Z":"20191031-145253-21773e69"}]', 'e970cc21-e176-4315-9e05-671468efab5b')
                        was aborted: ERROR: duplicate key value violates unique constraint "nodes_info_pkey"

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
I can split the PR in two, one for the read, that don't need to be atomic, and post-pone the writing part that need atomicity

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**